### PR TITLE
[cmake][module] Findpython add requirement for lzma for kodi_dependsbuild

### PR DESCRIPTION
**Pull request by fuzzard**
## Description
Intermittent linker failure due to lzma lib not being included in linker flags
```
Undefined symbols for architecture x86_64:
  "_lzma_alone_decoder", referenced from:
      __lzma_LZMADecompressor___init__ in libpython3.9.a(_lzmamodule.o)
  "_lzma_alone_encoder", referenced from:
      _Compressor_init in libpython3.9.a(_lzmamodule.o)
  "_lzma_auto_decoder", referenced from:
      __lzma_LZMADecompressor___init__ in libpython3.9.a(_lzmamodule.o)
  "_lzma_check_is_supported", referenced from:
      __lzma_is_check_supported in libpython3.9.a(_lzmamodule.o)
  "_lzma_code", referenced from:
      _compress in libpython3.9.a(_lzmamodule.o)
      __lzma_LZMADecompressor_decompress in libpython3.9.a(_lzmamodule.o)
  "_lzma_easy_encoder", referenced from:
      _Compressor_init in libpython3.9.a(_lzmamodule.o)
  "_lzma_end", referenced from:
      _Compressor_dealloc in libpython3.9.a(_lzmamodule.o)
      _Decompressor_dealloc in libpython3.9.a(_lzmamodule.o)
  "_lzma_get_check", referenced from:
      __lzma_LZMADecompressor_decompress in libpython3.9.a(_lzmamodule.o)
  "_lzma_lzma_preset", referenced from:
      _lzma_filter_converter in libpython3.9.a(_lzmamodule.o)
      _Compressor_init in libpython3.9.a(_lzmamodule.o)
  "_lzma_properties_decode", referenced from:
      __lzma__decode_filter_properties in libpython3.9.a(_lzmamodule.o)
  "_lzma_properties_encode", referenced from:
      __lzma__encode_filter_properties in libpython3.9.a(_lzmamodule.o)
  "_lzma_properties_size", referenced from:
      __lzma__encode_filter_properties in libpython3.9.a(_lzmamodule.o)
  "_lzma_raw_decoder", referenced from:
      __lzma_LZMADecompressor___init__ in libpython3.9.a(_lzmamodule.o)
  "_lzma_raw_encoder", referenced from:
      _Compressor_init in libpython3.9.a(_lzmamodule.o)
  "_lzma_stream_decoder", referenced from:
      __lzma_LZMADecompressor___init__ in libpython3.9.a(_lzmamodule.o)
  "_lzma_stream_encoder", referenced from:
      _Compressor_init in libpython3.9.a(_lzmamodule.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

## Motivation and context
@ksooo reported a linker issue that appears to be intermittent. Looking into it, for tools/depends builds, we statically compile the lzma module into our pythonlib, therefore we need to declare the lzma library to our linker.

## How has this been tested?
cmake generation for macos x86_64 target. Explicitly includes lzma lib in python_libraries variable now

## What is the effect on users?
N/A

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by fuzzard** [comment]
Jenkins build this please

File: cmake/modules/FindPython.cmake [hunk 1]
```diff
@@ -63,6 +63,7 @@ if(KODI_DEPENDSBUILD)
   find_library(EXPAT_LIBRARY expat REQUIRED)
   find_library(INTL_LIBRARY intl REQUIRED)
   find_library(GMP_LIBRARY gmp REQUIRED)
+  find_library(LIBLZMA_LIBRARY lzma REQUIRED)
```
Comment (wsnipex) [inline cmake/modules/FindPython.cmake:66]:
```suggestion
  find_library(LZMA_LIBRARY lzma REQUIRED)
```

**Review by wsnipex — APPROVED** [review]

**Comment by fuzzard** [comment]
Jenkins build and merge

**Comment by fuzzard** [comment]
unrelated flakey linux test failure. will merge anyway
```
04:40:08 [  FAILED  ] TestHTTPDirectory.LighttpDefaultIndex
```

**Pull request merged**